v1/sensor3700.js

```diff
@@ -39,158 +165,42 @@ module.exports = function (RED) {
-        const objectsList = decodeTLV(buffer, node);
-        const state = objectsList[0].getFloatValue();
-        const msg = {};
-        msg.payload = {};
-        msg.payload.state = node.state;
-        msg.payload.data = {
-          activeEnergy: state,
-        };
-        node.cache.activeEnergy = state;
-        msg.payload.cache = node.cache;
-        node.send(msg);
-      }).then((data) => {
-        node.send(data);
-      }).catch((err) => {
-        const msg = {};
-        msg.error = err;
-        node.send(msg);
-      });
-    }
-
-    if (node.reactivePower) {
-      node.device.observe('/3305/0/5815', (err, resp) => {
-        const buffer = Buffer.from(resp, 'base64');
-        const objectsList = decodeTLV(buffer, node);
-        const state = objectsList[0].getFloatValue();
-        const msg = {};
-        msg.payload = {};
-        msg.payload.state = node.state;
-        msg.payload.data = {
-          reactivePower: state,
-        };
-        node.cache.reactivePower = state;
-        msg.payload.cache = node.cache;
-        node.send(msg);
-      }).then((data) => {
-        node.send(data);
-      }).catch((err) => {
-        const msg = {};
-        msg.error = err;
-        node.send(msg);
-      });
-    }
-    if (node.reactiveEnergy) {
-      node.device.observe('/3305/0/5810', (err, resp) => {
-        const buffer = Buffer.from(resp, 'base64');
-        const objectsList = decodeTLV(buffer, node);
-        const state = objectsList[0].getFloatValue();
-        const msg = {};
-        msg.payload = {};
-        msg.payload.state = node.state;
-        msg.payload.data = {
-          reactiveEnergy: state,
-        };
-        node.cache.reactiveEnergy = state;
-        msg.payload.cache = node.cache;
-        node.send(msg);
-      }).then((data) => {
-        node.send(data);
-      }).catch((err) => {
-        const msg = {};
-        msg.error = err;
-        node.send(msg);
-      });
-    }
-    if (node.relay) {
-      node.device.observe('/3312/0/5850', (err, resp) => {
-        const buffer = Buffer.from(resp, 'base64');
-        const objectsList = decodeTLV(buffer, node);
-        const state = objectsList[0].getBooleanValue();
-        const msg = {};
-        msg.payload = {};
-        msg.payload.state = node.state;
-        msg.payload.data = {
-          relay: state,
-        };
-        node.cache.relay = state;
-        msg.payload.cache = node.cache;
+    node.device.getObjects().then(() => {
+      const msg = {};
+      msg.payload = `[Sensor3700-${node.device.id}] Sensor is already registered`;
+      node.send(msg);
+      node.state = true;
+      configure();
+    }).catch((err) => {
+      const msg = {};
+      if (err === 404) {
+        msg.payload = `[Sensor3700-${node.device.id}] Sensor is not yet registered. Waiting for registration event...`;
         node.send(msg);
-      });
-    }
-
-    node.device.write('/1/0/3', () => {
```

Comment:
Do not forget to add observation interval writing to configure method! (copy paste from new version)